trunner/device.py

```diff
@@ -374,13 +389,52 @@ def reset(self):
         time.sleep(0.050)
         self.reset_gpio.high()
 
-    def boot(self, serial_downloader=False):
+    def restart(self):
+        # without powering down rpi ports eval kit is partially powered through usb
+        rpi_ports_off = subprocess.run([
+            'uhubctl',
+            '-l', '2',
+            '-a', '0'],
+            stdout=subprocess.DEVNULL
+        )
+        if rpi_ports_off.returncode != 0:
+            logging.error('uhubctl failed!\n')
+            raise Exception('RPi ports power down failed!')
+
+        self.restart_gpio.low()
+        time.sleep(0.500)
+        self.reset_gpio.low()
+        self.restart_gpio.high()
+        time.sleep(0.500)
+        self.reset_gpio.high()
+        rpi_ports_on = subprocess.run([
+            'uhubctl',
+            '-l', '2',
+            '-a', '1'],
+            stdout=subprocess.DEVNULL
+        )
+        if rpi_ports_on.returncode != 0:
+            logging.error('uhubctl failed!\n')
+            raise Exception('RPi ports power up failed!')
+
+        # after power on there is pre-restart device, which will disappear
+        wait_for_disappear(DEVICE_SERIAL, timeout=5)
```

Comment:
What if device won't disappear?

Reply:
If device won't disappear after 5 seconds it means that's the right device and wait_for_dev is executed (which doesn't wait, because dev is ready)

Reply:
Thanks, I will test using /sys/bus/usb/drivers/usb/unbind and bind